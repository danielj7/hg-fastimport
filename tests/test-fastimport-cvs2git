#!/bin/sh

# Test a dump of the style created by cvs2git.  Notable features:
#   - the dump is in two files: all the file data (blobs) in one,
#     and everything else (i.e. commit metadata) in the other

# The initial CVS repository was created like this:
#   mkdir t && cd t
#   export CVSROOT=$PWD/repo
#   cvs init
#   mkdir repo/proj
#   cvs co proj
#   cd proj 
#   echo "hello there" > foo.txt
#   echo "ooga booga" > ooga
#   cvs add foo.txt ooga 
#   cvs ci -m"initial revision"
#   echo "woof woof\nooga booga\nblah blah" > ooga
#   cvs ci -m"modify"
#
# Then I generated the dumps with
#   cvs2git --options cvs2git.options
#
# where cvs2git.options was a copy of cvs2git-example.options
# from the trunk of cvs2svn, modified to point the CVS repo
# I just created.

. $TESTDIR/fastimport-common

cat > git-blob.dat <<EOF
blob
mark :1
data 12
hello there

blob
mark :2
data 31
woof woof
ooga booga
blah blah

blob
mark :3
data 11
ooga booga

EOF

cat > git-dump.dat <<EOF
commit refs/heads/master
mark :1000000000
committer example <example> 1200000000 +0000
data 17
initial revision

M 100644 :1 foo.txt
M 100644 :3 ooga

commit refs/heads/master
mark :1000000001
committer example <example> 1200000060 +0000
data 7
modify

M 100644 :2 ooga

EOF

echo "% import cvs2git dumps"
hg init simplecvs
hg -R simplecvs fastimport git-blob.dat git-dump.dat \
  | sed "s|$HGTMP|HGTMP|g"

# Inspecting the hg repository is identical to test-fastimport-git.
echo "% log"
cd simplecvs
hg glog -v

echo "% manifest at rev 0"
hg manifest -r0

echo "% manifest at rev 1"
hg manifest -r1

echo "% cat foo.txt at rev 0"
hg cat -r0 foo.txt
echo "% cat foo.txt at rev 1"
hg cat -r1 foo.txt

echo "% cat ooga at rev 0"
hg cat -r0 ooga
echo "% cat ooga at rev 1"
hg cat -r1 ooga
