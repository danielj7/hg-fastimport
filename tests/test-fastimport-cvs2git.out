% import cvs2git dumps
scanning source...
sorting...
converting...
1 initial revision
0 modify
Removing blob dir 'HGTMP/test-fastimport-cvs2git/simplecvs/.hg/blobs' ...
% log
o  changeset:   1:f5f5b99690f9
|  tag:         tip
|  user:        example
|  date:        Thu Jan 10 21:21:00 2008 +0000
|  files:       ooga
|  description:
|  modify
|
|
o  changeset:   0:602b297b695e
   user:        example
   date:        Thu Jan 10 21:20:00 2008 +0000
   files:       foo.txt ooga
   description:
   initial revision


% manifest at rev 0
foo.txt
ooga
% manifest at rev 1
foo.txt
ooga
% cat foo.txt at rev 0
hello there
% cat foo.txt at rev 1
hello there
% cat ooga at rev 0
ooga booga
% cat ooga at rev 1
woof woof
ooga booga
blah blah
